### DISCUSSION TITLE
Best way to run (Host/Tenant) Jobs from Quartz?

### DISCUSSION THREAD

**Discussion by ShaneCourtrille** [OrchardCMS/OrchardCore#13350] [Q&A]
I haven't seen any module related to Quartz but we've decided to move our jobs over to there for easier/better management (we don't use the OC Admin UI at all -- can't even get to it actually anymore).  Is anyone doing that already and can give advice on the best way to do it?  I'm thinking we may want to expose the functionality via an API so they can still run within the right OC scope/context.

**Comment by hishamco** [comment]
OC already have background tasks, may be Quartz has some features that are not supported yet in OC, @jtkech could you please clarify what we have and what are the limitations?

/cc @lahma

**Reply by lahma** [reply]
There was some [discussion about Quartz features earlier](https://github.com/OrchardCMS/OrchardCore/issues/760). Basically Quartz has more features that might not benefit the "normal" use case in OC, but could give added value as an add-on. Quartz supports MS DI so anything coming from there should work, or if needed, a custom job factory injecting more context might be in order.
